client/me/profile-links/data/types.ts

```diff
@@ -0,0 +1,31 @@
+import { UseMutationOptions } from '@tanstack/react-query';
+
+export type ProfileLink = {
+	link_slug: string;
+	title: string;
+	value: string;
+};
+
+export type AddProfileLinkPayload = Pick< ProfileLink, 'title' | 'value' >;
+
+export type AddProfileLinksPayload = AddProfileLinkPayload[];
+
+export type ProfileLinkResponse = ProfileLink[];
```

Comment:
Some of these types seem arbitrary and could be inlined at no cost. `ProfileLink[]` is even shorter than `ProfileLinkResponse`.